From d086421ca478940b8588e33c9f836472a578b199 Mon Sep 17 00:00:00 2001
From: Oleg Oshmyan <chortos@inbox.lv>
Date: Sat, 11 Jul 2020 21:06:48 +0300
Subject: [PATCH] osd_libass: set ScaledBorderAndShadow

libass recently switched the default from 1 to 0 for compatibility
with ASS scripts that rely on the historical/VSFilter default of 0.

libass does attempt to detect and avoid breaking scripts that rely
on the historic libass-only default of 1, but it doesn't cover tracks
created directly through the API, so set the header explicitly.

Fixes https://github.com/mpv-player/mpv/issues/7900.
---
 sub/osd_libass.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sub/osd_libass.c b/sub/osd_libass.c
index 2efcef00be..5f9bc52f66 100644
--- a/sub/osd_libass.c
+++ b/sub/osd_libass.c
@@ -132,6 +132,7 @@ static void create_ass_track(struct osd_state *osd, struct osd_object *obj,
     track->Timer = 100.;
     track->WrapStyle = 1; // end-of-line wrapping instead of smart wrapping
     track->Kerning = true;
+    track->ScaledBorderAndShadow = true;
 
     update_playres(ass, &obj->vo_res);
 }
